src/main/java/org/sonar/plugins/cas/session/SessionFileRemover.java

```diff
@@ -0,0 +1,196 @@
+        List<Path> ticketsToBeRemoved = new ArrayList<>();
+        List<String> expiredJwtIdsCopy = new ArrayList<>(expiredJwtIds);
+
+        for (Path ticketFile : serviceTicketFiles) {
+            filterServiceTicketWithJwtIds(ticketsToBeRemoved, expiredJwtIdsCopy, ticketFile);
+        }
+
+        return ticketsToBeRemoved;
+    }
+
+    private void filterServiceTicketWithJwtIds(List<Path> ticketsToBeRemoved, List<String> jwtIds, Path ticketFile) {
+        String ticketFileRaw = ticketFile.toString();
+        String jwtIdToBeCompared;
+
+        try {
+            byte[] fileContent;
+            fileContent = Files.readAllBytes(ticketFile);
+            jwtIdToBeCompared = new String(fileContent, StandardCharsets.US_ASCII).trim();
+        } catch (IOException e) {
+            LOG.error("Could not filter Service Ticket " + ticketFileRaw + " for removal", e);
+            return;
+        }
+
+        if (jwtIds.contains(jwtIdToBeCompared)) {
+            LOG.debug("Found Service Ticket {} for removal", ticketFileRaw);
+            ticketsToBeRemoved.add(ticketFile);
+            // reduce search complexity
+            jwtIds.remove(jwtIdToBeCompared);
+        }
+    }
+
+    Collection<Path> listAllServiceTickets() {
+        Stream<Path> list = listSessionStorePath();
+        return list.filter(file -> file.getFileName().toString().startsWith(SERVICE_TICKET_PREFIX))
+                .collect(Collectors.toList());
+    }
+
+    private Stream<Path> listSessionStorePath() {
+        Path dir = Paths.get(sessionStorePath);
+        try {
+            return Files.list(dir);
+        } catch (IOException e) {
+            throw new RuntimeException("Could not list files in CAS file session store", e);
+        }
+    }
+
+    void removeServiceTickets(String... serviceTicketFileNames) {
+        for (String serviceTicketFileName : serviceTicketFileNames) {
+            removeServiceTicket(serviceTicketFileName);
+        }
+    }
+
+    private void removeServiceTicket(String serviceTicketFileName) {
+        try {
+            Path file = Paths.get(sessionStorePath, serviceTicketFileName);
+            Files.delete(file);
+        } catch (IOException e) {
+            LOG.error("Could not delete Service Ticket file from file session store", e);
+        }
+    }
+
+    Collection<Path> listAllJwtFiles() {
+        return listSessionStorePath()
+                .filter(file -> JWT_FILE_PATTERN.matcher(file.getFileName().toString()).matches())
+                .collect(Collectors.toList());
+    }
+
+    List<SimpleJwt> parseJwtFiles(Collection<Path> foundFiles) {
+        List<SimpleJwt> list = new ArrayList<>(foundFiles.size());
+
+        for (Path file : foundFiles) {
+            try (BufferedReader reader = Files.newBufferedReader(file, StandardCharsets.US_ASCII)) {
+                SimpleJwt jwt = JAXB.unmarshal(reader, SimpleJwt.class);
```

Comment:
JAXB throws a `RuntimeException` if the file could not be parsed. The current implementation looks like if one file could not be parsed, the session store gets never cleaned. I think we could catch the JAXB exception too.

Reply:
Catches now all exceptions and should not break any more, and produces an error log.